### DISCUSSION TITLE
Optimize feature

### DISCUSSION THREAD

**Discussion by phancddev** [engageub/InternetIncome#83] [Ideas]
I think you can use another way to connect proxy (like use iptable) or you can optimize container proxy, remove any layer of container.
Thanks for reading!

**Comment by engageub** [comment]
Hello,
Thank you for the suggestion, please note that proxy and multi IP are different.
For multi IP support, please see the test branch. It does not have extra layer of container. 
Also, note that each docker image application has its own flavour of Linux version.
Any modifications are to be done before starting the container.
All the applications are to be downloaded manually and a separate docker image has to be built and a proxy can be given similar to how network proxy is setup in ubuntu gnome desktop.
If you are aware of how to accomplish, please provide a working example. 

https://github.com/engageub/InternetIncome/tree/test


Thank you

**Comment by phancddev** [comment]
I mean use container proxy is so heavy with system. Someone use another
ways to forward proxy and its so optimize. I don’t know how but maybe they
said they use redsocks, proxychains with iptable to redirect proxy. Or
optimize container proxy to lite

**Reply by engageub** [reply]
Thank you for the suggestion. Redsocks and proxychains have already been tried before finalizing the code.
Chatgpt or other AI tools provided these suggestions but they did not work.

Thank you

**Comment by phancddev** [comment]
Thank you for rely!
Its very great if you can optimize proxy containers. Thanks!!!

**Reply by engageub** [reply]
proxychains does not work with all the apps. An issue has already been raised to them. Packetstream works with environment variables of HTTP_PROXY , so it can work without extra layer.
Not all applications are designed to pick up values from environment variables.
If you can provide a working example, that would be good to know for implementation. 
Thank you

**Comment by phancddev** [comment]
Oh, nice

Because starting a separate proxy container for each app container is quite
heavy and because the proxy container is also heavy, it causes the system
to be unoptimized. If there is a way to make the proxy container have
almost no impact on the system, that would be great!!

**Reply by engageub** [reply]
It does not start proxy container for each app.
It starts one container only for one proxy. All the applications are run on that single proxy container.

Thank you

**Reply by engageub** [reply]
May I know if you are creating multiple folders for multiple proxies?
Its not required. All proxies are to be added in proxies.txt
Please run the following command to see the number of proxy containers and number of apps. 

```
sudo docker container ls
```

**Comment by phancddev** [comment]
Yes its in 1 file proxy.txt. But u knew if one container app open -> one
container proxy open

**Reply by engageub** [reply]
No.
It starts the proxy container first.
Then all the apps are started on that network.
Then the next proxy container starts.
Then the apps corresponding to that network
......

So, only 1 proxy container for 1 proxy and not app.

Thank you
